src/extractors.rs

```diff
@@ -100,27 +108,27 @@ fn section<'a, I>(events: &mut I) -> Result<DocSection, ParseError> where
             DocSection::LifetimeParameters(try!(list(events, &headline))),
         "returns" =>
             DocSection::Returns(
-                md(events.take_while(not!(start Tag::List(_)))),
+                md(events.map(|(event, _)| event).take_while(not!(start Tag::List(_)))),
                 try!(list(events, &headline))),
         _ =>
             DocSection::Custom(
                 headline,
-                md(events.take_while(not!(start Tag::Header(1))))),
+                md(events.map(|(event, _)| event).take_while(not!(start Tag::Header(1))))),
```

Comment:
Yeah, well, implementing a custom map specialization is nice, but probably much more work and not that much shorter to type in the end.